metaflow/runtime.py

```diff
@@ -698,10 +675,100 @@ def __init__(
         self.datastore_sysroot = flow_datastore.datastore_root
         self._results_ds = None
 
+        origin = None
         if clone_run_id and may_clone:
-            self._is_cloned = self._attempt_clone(clone_run_id, join_type)
+            origin = self._find_origin_task(clone_run_id, join_type)
+        if origin and origin["_task_ok"]:
+            # At this point, we know we are going to clone
+            self._is_cloned = True
+            if reentrant:
```

Comment:
Can we add a comment here to note the expected reentrant behavior?

Reply:
Sure, would this work:
```
If `reentrant`, multiple processes may be trying to "clone" at the same time. In this case, we want only one to perform the actual clone and the others to wait for the clone to be done prior to proceeding.
```